soci/gzip_zinfo.go

```diff
@@ -28,6 +28,24 @@ import (
 	"unsafe"
 )
 
+const windowSize = 32768
+
+// GzipIndexPoint corresponds to gzip_index_point in indexer.c
+type GzipCheckPoint struct {
+	Out    int64  /* corresponding offset in uncompressed data */
+	In     int64  /* offset in input file of first full byte */
+	Bits   int8   /* number of bits (1-7) from byte at in - 1, or 0 */
+	Window []byte /* preceding 32K of uncompressed data */
+}
+
+// GzIndex corresponds to gzip_index in indexer.c
+type CZinfo struct {
+	Have     int64            /* number of list entries filled in */
+	Size     int64            /* number of list entries allocated */
```

Comment:
Yes, because we want to precisely check the contents of gzip_checkpoint. This is used in tests only.

Reply:
We can technically remove one of these, since `have` will always equal `size`.

Reply:
While we have the structure defined in zinfo.h in precisely that way, I would still prefer keeping as is.

Reply:
Can you explain the rationale for your preference? Generally speaking, test-only methods and data structures are a code smell.

Reply:
This code has no use in production. The only reason for it to exist is to be able to do `deep equal` of the gzip's Checkpoints collection. It may make more sense to move it back to testutils. 

Yes, I agree that this is a code smell, but the alternative is to directly access C struct in unit tests and do those comparisons, which doesn't seem to be worth it.

Reply:
Removed the unmarshal and replaced it with returning the index of the first different byte in sequence.